language-design.md

```diff
@@ -38,6 +181,25 @@ How do the themes of _Growing a Language_ relate to the "sound lab" we did this
 
 **Response**
 
+One of the key points that Steele emphasizes is to “think of a language design as a pattern 
+for language designs, a tool for making more tools of the same kind” [Steele, 1998]. So a well-designed 
+language ideally has a solid pattern for growth and allows for new functionalities to be added with 
+relative ease. For the sound lab, one of the main design flaws was the fact that it was tedious for users 
+to make multiple sound modifications on a single audio file. By addressing this problem, we explored the 
+ways in which we could implement a new sort of pattern for combining functions in the library. One way was 
+a functional programming approach, where functions would be composed to make multiple modifications, whereas
+another way was a more object oriented approach, where functions would be chained together in the order the 
+user wants them to be called. In this situation, deciding which approach to take is a really important design
+decision to make because it’ll determine the pattern by which the user is going to be able to use existing 
+methods as well as ones that may be added in the future.
+
+Another overarching theme of Steele’s paper was the idea of using a very minimal vocabulary and being able 
```

Comment:
I came to a similar conclusion when redesigning the sound lab.  You isolated what I also considered to be the two main failings of the original sound lab, redundancy and a lack of fluency.